### PR TITLE
Small Refactor and Adds Amplitude Event to Email Send Button

### PR DESCRIPTION
**Pull request by hannahbergam**
This PR handles a small refactor (namely, uses a passed through variable for isSignedIn instead of hooking up to current user redux). The translation/text constants are updated accordingly. 

Then, it sets up some scaffolding for the send email button, splitting into a function for saving the inputs and sending the email. The email send function will also send the amplitude event, which is now set up!

Screenshot 2023-10-24 at 4 07 01 PM

## Links

- spec: https://docs.google.com/document/d/1jSmRkX8O-PMDj9y-hvMubOMl1hkcRz8o9_nUO1s7gls/edit?pli=1
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-1078

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/util/AnalyticsConstants.js [hunk 1]
```diff
@@ -86,7 +86,9 @@ const EVENTS = {
   TA_RUBRIC_ON_STUDENT_WORK_UNLOADED: 'TA Rubric On Student Work Unloaded',
   TA_RUBRIC_SUBMITTED: 'TA Rubric Submitted',
 
+  // Hour of Code
   AGE_21_SELECTED_EVENT: 'Age 21+ Selected',
+  GUIDE_SENT_EVENT: 'Guide Sent',
```
Comment (TurnerRiley) [inline apps/src/lib/util/AnalyticsConstants.js:91]: I searched Amplitude for an event with this name but didn't find one, do we have one set up for this?

**Review by TurnerRiley — APPROVED** [review]
Looks great! Just had a question about the event on Amplitude

File: apps/src/lib/util/AnalyticsConstants.js (hunk 1 above)
Comment (dmcavoy) [reply apps/src/lib/util/AnalyticsConstants.js:91]: I can add it now.

Comment (dmcavoy) [reply apps/src/lib/util/AnalyticsConstants.js:91]: Done!

Comment (hannahbergam) [reply apps/src/lib/util/AnalyticsConstants.js:91]: Thanks Dani!

**Pull request merged**